src/coord/src/sink_connector.rs

```diff
@@ -97,7 +245,14 @@ async fn build_kafka(
     frontier: Antichain<Timestamp>,
     id: GlobalId,
 ) -> Result<SinkConnector, CoordError> {
-    let topic = format!("{}-{}-{}", builder.topic_prefix, id, builder.topic_suffix);
+    let topic = if builder.exactly_once {
+        format!("{}-{}", builder.topic_prefix, id)
```

Comment:
So I think there's a subtle flaw here. My understanding is that if someone makes a sink with their topic name set to `mytopic` and lets say they get bound to globalid `u1` - then the first time they do this they get back a new sink called `mytopic-u1`. awesome. but then ... if they blow away their catalog, run things again, define a new view and a new sink going to `mytopic` and lets say they get bound to `u1` again because of the catalog deletion, then I'm not totally sure what happens (I think they might fail if the schema registry conflicts and otherwise they might succeed)

Reply:
I think we could defensibly choose to ignore that failure mode for now but I think we could maybe address it and make the code simpler because it seems like theres two modes here

- the first time we create a sink -- here both the exactly-once sinks and the current status quo sinks need to make a new topic and register new schemas and fail if thats not the case

- on restart - exactly once sinks need to find an existing topic / schema and current status quo sinks need to create a new topic / register new schemas

and it seems like we might be able to violate both "needs" for exactly once sinks here?

the "create for the first time" vs "restart" workflows (thingys?) correspond to two distinct callsites for `sink_connector::build` - I think it maybe makes sense to pass in a `is_restart` boolean and then branch based on that to either `register_kafka_topic` or `reload_kafka_topic` or something like that. My hope is that this can make the code simpler by clearly delineating what work is done on restart (and what invariants need to hold on restart) vs what work is done at creation time (and what invariants need to hold at creation time).

I think its also legit to merge this more or less as is and then iterate, especially if its all hardcoded to off

Reply:
This makes sense and I think that refactor will fix it and make it cleaner. I'm going to commit this current PR if CI passes, then work on this fix in a follow-up PR - I'm getting too bogged down in rebasing/keeping this PR up with main at my current comparatively low productivity